pkg/manager/chemanager_controller.go

```diff
@@ -31,21 +32,36 @@ import (
 )
 
 var (
-	log = ctrl.Log.WithName("che")
+	log             = ctrl.Log.WithName("che")
+	currentManagers = map[client.ObjectKey]v1alpha1.CheManager{}
+	managerAccess   = sync.Mutex{}
 )
 
 type CheReconciler struct {
 	client  client.Client
 	scheme  *runtime.Scheme
 	gateway gateway.CheGateway
-	syncer  sync.Syncer
+	syncer  datasync.Syncer
+}
+
+func GetCurrentManagers() map[client.ObjectKey]v1alpha1.CheManager {
```

Comment:
can you please explain to me why we need multiple managers and why we need to do it like this with mutex?

Reply:
Added the docs about what this is for and its pros and cons.